server/src/main/java/org/elasticsearch/index/mapper/DateFieldMapper.java

```diff
@@ -213,21 +213,38 @@ public Builder(String name, Resolution resolution, DateFormatter dateFormatter,
         }
 
         protected DateFieldType setupFieldType(BuilderContext context) {
-            DateFormatter dateTimeFormatter = DateFormatter.forPattern(format.getValue()).withLocale(locale.getValue());
-            return new DateFieldType(buildFullName(context), index.getValue(), docValues.getValue(),
-                dateTimeFormatter, resolution, meta.getValue());
+            try {
+                DateFormatter dateTimeFormatter = DateFormatter.forPattern(format.getValue()).withLocale(locale.getValue());
+                return new DateFieldType(buildFullName(context), index.getValue(), docValues.getValue(),
+                    dateTimeFormatter, resolution, meta.getValue());
+            }
+            catch (IllegalArgumentException e) {
+                throw new IllegalArgumentException("Error parsing [format] on field [" + name() + "]: " + e.getMessage(), e);
+            }
         }
 
         @Override
         protected List<Parameter<?>> getParameters() {
             return List.of(index, docValues, store, format, locale, nullValue, ignoreMalformed, boost, meta);
         }
 
+        private Long parseNullValue(DateFormatter formatter) {
+            if (nullValue.getValue() == null) {
+                return null;
+            }
+            try {
+                return formatter.parseMillis(nullValue.getValue());
+            }
+            catch (Exception e) {
```

Comment:
Small comment, catch is usually on same line as previous brace.

Not relevant for this PR, but reading this made me notice that we parse `null_value` dates different from ones that appear in documents. Specifically, I wonder why we don't use `DateFieldType#parse` here ?

Reply:
> Specifically, I wonder why we don't use DateFieldType#parse here ?

This is a bug, I'll open a separate PR to fix.